### PR TITLE
in rest api do not return unwrapped then deleted object keys, and mor…

### PR DESCRIPTION
**Pull request by longbowlu**
## Description 

Wrapped objects are not stored in authority state so we can't fetch it.

## Test Plan 

deployed to one FN and fixed the issue

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes
in rest api do not return unwrapped then deleted object keys

### CODE REVIEW

**Review by gegaowp — APPROVED** [review]

File: crates/sui-rest-api/src/checkpoints.rs [hunk 1]
```diff
@@ -100,13 +106,24 @@ pub async fn get_full_checkpoint(
             )
             .collect::<HashSet<_>>()
             .into_iter()
+            // Unwrapped-then-deleted objects are not stored in state before the tx, so we have nothing to fetch.
```
Comment (amnn) [inline crates/sui-rest-api/src/checkpoints.rs:109]: This is not an issue in this PR (so not blocking), but just noting that the pattern of collecting into a `HashSet` and then draining that into a `Vec` makes me nervous from a determinism perspective -- i.e. the response order could change non-deterministically -- is there a reason we can't use a `BTreeSet` instead here?

**Review by amnn — APPROVED** [review]

**Review by bmwill — APPROVED** [review]

File: crates/sui-rest-api/src/checkpoints.rs (hunk 1 above)
Comment (longbowlu) [reply crates/sui-rest-api/src/checkpoints.rs:109]: that's a valid point but i don't think the order matters here

Comment (amnn) [reply crates/sui-rest-api/src/checkpoints.rs:109]: I agree, order doesn't matter, but what I'm more concerned about is that having a stable order can be important to make testing easier, etc.

Comment (bmwill) [reply crates/sui-rest-api/src/checkpoints.rs:109]: that's actually a good point re testability.

**Pull request merged**